src/main/java/com/fasterxml/jackson/databind/jsontype/impl/AsDeductionTypeDeserializer.java

```diff
@@ -41,9 +41,9 @@ public class AsDeductionTypeDeserializer extends AsPropertyTypeDeserializer
     private final Map<BitSet, String> subtypeFingerprints;
 
     public AsDeductionTypeDeserializer(JavaType bt, TypeIdResolver idRes, JavaType defaultImpl,
-            DeserializationConfig config, Collection<NamedType> subtypes)
+            DeserializationConfig config, Collection<NamedType> subtypes, boolean hasTypeResolverAnnotation)
```

Comment:
could you add back the old api method and hardcode its hasTypeResolverAnnotation (to false, perhaps)? -- this would prevent a backward compatibility issue.

could you add Javadoc to the new constructor with `@since 2.15`